cns/restserver/api_test.go

```diff
@@ -1292,6 +1292,37 @@ func TestCreateHostNCApipaEndpoint(t *testing.T) {
 	fmt.Printf("createHostNCApipaEndpoint Responded with %+v\n", createHostNCApipaEndpointResponse)
 }
 
+func TestNCIDCaseSensitive(t *testing.T) {
+	setEnv(t)
+	err := setOrchestratorType(t, cns.Kubernetes)
+	if err != nil {
+		t.Fatalf("TestNCIDCaseSensitive failed with error:%+v", err)
+	}
+
+	// test both upper-case NCIDs and cns-managed mode NCID
+	upperCaseNCID1 := "Swift_89063DBF-AA31-4BFC-9663-3842A361F188"
+	upperCaseNCID2 := "Swift_89063DBF-AA31-4BFC-9663-3842A361F189"
+	managedModeNCID := "89063DBF-AA31-4BFC-9663-3842A361F188"
+	upperCaseNCIDs := []string{upperCaseNCID1, upperCaseNCID2, managedModeNCID}
+	nmaProgrammedNCVersionStr1 := "1"
+	nmaProgrammedNCVersionStr2 := "2"
+	nmaProgrammedNCVersionStr3 := "3"
+	ncVersionList := map[string]string{}
+
+	// add lower-case NCIDs to ncVersionList
+	ncVersionList["Swift_89063dbf-aa31-4bfc-9663-3842a361f188"] = nmaProgrammedNCVersionStr1
```

Comment:
instead of hardcoding the lower case values as keys in ncVersionList. can you use the func that you had created to convert to lower case?